### PR TITLE
Fix/truncate

### PR DESCRIPTION
**Pull request by olcplif**
In the previous version, text truncating could happen in the middle of an open tag image or video, which led to incorrect behavior. This patch is designed to fix this bug.
[https://trello.com/c/1b5q0WU3/79-fix-ckeditor](https://trello.com/c/1b5q0WU3/79-fix-ckeditor)
- [x]  fix truncate text in index pages

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -41,5 +41,5 @@
 yarn-debug.log*
 .yarn-integrity
 /vendor
-/coverage/*
+/coverage/
```
Comment (mmrshk) [inline .gitignore:44]: `coverage/*`

File: app/presenters/article_presenter.rb [hunk 2]
```diff
@@ -1,11 +1,42 @@
...
     @article.tags.present? ? @article.tags.pluck(:name).join(', ') : ''
   end
+
+  def truncate_article_text
+    truncate(@article.text.to_s, escape: false, length: length_truncate) do
+      link_to 'Continue', controller: 'articles', action: 'show', id: @article.id
+    end
+  end
+
+  REGEX_FIGURE = /<figure[^>]/
```
Comment (mmrshk) [inline app/presenters/article_presenter.rb:22]: constants always locates after includes in the beginning of the file

Comment (mmrshk) [reply app/presenters/article_presenter.rb:22]: have you ran here rubocop?

File: .gitignore (hunk 1 above)
Comment (olcplif) [reply .gitignore:44]: fixed

File: app/presenters/article_presenter.rb (hunk 2 above)
Comment (olcplif) [reply app/presenters/article_presenter.rb]: thanks. fixed

File: app/presenters/article_presenter.rb [hunk 3]
```diff
@@ -1,11 +1,42 @@
...
 
 class ArticlePresenter
+  include ActionView::Helpers
+  include Rails.application.routes.url_helpers
+  include ActionView::Context
+
+  TAG_FIGURE_REGEX = /<figure[^>]/
+  TEXT_WITH_FIGURE_REGEX = %r{[\s\S]*?</figure>}
+  LENGTH_TRUNCATE_DEFAULT = 500
+  LENGTH_ADJUSTMENT_TRUNCATE = 0
```
Comment (mmrshk) [inline app/presenters/article_presenter.rb:11]: 0?

**Review by mmrshk — APPROVED** [review]
lgtm

**Pull request merged**